script.js

```diff
@@ -1,4 +1,80 @@
-var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
-};
+// variable assignments
+var scissors = "scissors";
+var paper = "paper";
+var stone = "stone";
+var reversedScissors = "reversed scissors";
+var reversedPaper = "reversed paper";
+var reversedStone = "reversed stone";
+var instructions =
+  'Type in "scissors" "paper" or "stone" to throw as your next hand. Good luck! <br>(Psst. You can add reversed before your hand to do inverted scissors papers stone!)';
+// RNG into computer response
+var randomHand = function () {
+  var randomNum = Math.floor(Math.random() * 3);
+  if (randomNum == 0) {
+    return scissors;
+  }
+  if (randomNum == 1) {
+    return paper;
+  }
+  if (randomNum == 2) {
+    return stone;
+  }
+};
+// game rules. surely room for improvement in efficiency?
+function matchOutcome(computerHand, userHand) {
+  if (
+    userHand == computerHand ||
+    (userHand == reversedStone && computerHand == stone) ||
+    (userHand == reversedPaper && computerHand == paper) ||
+    (userHand == reversedScissors && computerHand == scissors)
+  ) {
+    return "Draw!";
+  }
+  if (
+    (userHand == scissors && computerHand == paper) ||
+    (userHand == paper && computerHand == stone) ||
+    (userHand == stone && computerHand == scissors) ||
+    (userHand == reversedPaper && computerHand == scissors) ||
+    (userHand == reversedScissors && computerHand == stone) ||
+    (userHand == reversedStone && computerHand == paper)
+  ) {
+    return "You Win!";
+  }
+  if (
+    (userHand == scissors && computerHand == stone) ||
+    (userHand == paper && computerHand == scissors) ||
+    (userHand == stone && computerHand == paper) ||
+    (userHand == reversedPaper && computerHand == stone) ||
+    (userHand == reversedScissors && computerHand == paper) ||
+    (userHand == reversedStone && computerHand == scissors)
+  ) {
+    return "You Lost!";
+  }
+}
+// looked for missing brackets here for half the assignment time
+// result message output, included with every successful loop
+var resultMessage = function (computerHand, userHand) {
+  return (
+    "The computer chose " + computerHand + ". <br> You chose " + userHand + "."
+  );
+};
+// error message for unexpected inputs
+var main = function (input) {

```

Comment:
i like how you managed to pull out your logic intensive code to its own helper functions which makes the code more reusable and readable